### PR TITLE
Remove `MptConfig`  

### PR DESCRIPTION
**Pull request by megha95**
Reverting PR #1529 
Transformers' lib has a different version of `MptConfig`; As MPT version maintained by Mosaic has the latest version, all `MptConfig` refs are changed to `MPTConfig` in this PR
Note, even after this change, code will still be compatible with both versions (i.e with or without trust_remote_code).

### CODE REVIEW

File: vllm/transformers_utils/configs/mpt.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                 logit_scale: Optional[Union[float, str]] = None,
+                 no_bias: bool = False,
+                 embedding_fraction: float = 1.0,
+                 norm_type: str = 'low_precision_layernorm',
+                 use_cache: bool = False,
+                 init_config: Dict = init_config_defaults,
+                 fc_type: str = 'torch',
+                 verbose: Optional[int] = None,
+                 **kwargs: Any):
+        """The MPT configuration class.
```
Comment (WoosukKwon) [inline vllm/transformers_utils/configs/mpt.py:63]: Can we add something like `# pylint: disable=line-too-long` here to pass the formatter?

**Review by WoosukKwon — COMMENTED** [review]
Hi @megha95, thanks for submitting the PR! Could you check the format again?

File: vllm/transformers_utils/configs/mpt.py (hunk 1 above)
Comment (megha95) [reply vllm/transformers_utils/configs/mpt.py:65]: done

**Review by WoosukKwon — APPROVED** [review]
LGTM. Thanks for the PR!

FYI, I've made some minor stylistic changes.

**Pull request merged**